cmd/main_test.go

```diff
@@ -168,3 +168,83 @@ here"
 		}
 	}
 }
+
+func Test_isCaddyfile(t *testing.T) {
+	type args struct {
+		configFile  string
+		adapterName string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "local Caddyfile without adapter",
+			args: args{
+				configFile:  "./Caddyfile",
+				adapterName: "",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "local caddyfile with adapter",
+			args: args{
+				configFile:  "./Caddyfile",
+				adapterName: "caddyfile",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "ends with .caddyfile with adapter",
+			args: args{
+				configFile:  "./conf.caddyfile",
+				adapterName: "caddyfile",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "ends with .caddyfile without adapter",
+			args: args{
+				configFile:  "./conf.caddyfile",
+				adapterName: "",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "config is Caddyfile.yaml without adapter",
+			args: args{
+				configFile:  "./Caddyfile.yaml",
+				adapterName: "",
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "json is not caddyfile but not error",
```

Comment:
We can duke it out later, but might be worth discussing at some point.

Reply:
I'm using an error return as a signal of ambiguity. We consider `.json` a known extension, so it's not ambiguous. The determination is: .json is not ambiguous and isn't a Caddyfile.

Reply:
I use files like `Caddyfile.test2` kinda thing, it's not uncommon to have like `Caddyfile.dev` and `Caddyfile.prod`. I'm not sure I understand how those will behave from looking at the code (just woke up and still away from home 😅)

Reply:
If you pass the `--adpater`, it will go through as Caddyfile. If this is common, I'd rather we strip the whole `.caddyfile` detection out.

Reply:
I think it is common, and historically it has gone with the "starts with Caddyfile" rule. I think we only want to say it's ambiguous if there is a loaded adapter that matches the extension, not otherwise. I think it's fine if `Caddyfile.yaml` works if the yaml adapter isn't loaded. I realize that it's probably hard to test for though since we don't have any in Caddy itself. Maybe for testing, can make a dummy adapter?

Reply:
Then I think we should accept that users make silly mistakes such as passing `--config Caddyfile.yaml` without `--adapter`, or will do `caddy run` without specifying the `--adapter` which will cause the yaml file to be read as Caddyfil. It'll just fail silently (or loudly 🤷🏼‍♂️ ).